packages/trpc/server/routers/viewer/teams/get.handler.ts

```diff
@@ -37,15 +37,25 @@ export const getHandler = async ({ ctx, input }: GetOptions) => {
   // Hide Members of team when 1) Org is private and logged in user is not admin or owner
   // OR
   // 2)Team is private and logged in user is not admin or owner of team or Organization's admin or owner
-  const hideMembers =
-    (ctx.user.profile?.organization?.isPrivate && !ctx.user.organization?.isOrgAdmin) ||
-    (team.isPrivate &&
-      !(membership.role === MembershipRole.OWNER || membership.role === MembershipRole.ADMIN) &&
-      !ctx.user.organization?.isOrgAdmin);
+  function shouldHideMembers() {
+    const isOrgPrivate = ctx.user.profile?.organization?.isPrivate;
+    const isOrgAdminOrOwner = ctx.user.organization?.isOrgAdmin;
+    const isTargetingOrg = input.teamId === ctx.user.organizationId;
+
+    if (isTargetingOrg && isOrgPrivate && !isOrgAdminOrOwner) {
+      return true;
+    }
+    const isTeamAdminOrOwner =
+      membership?.role === MembershipRole.OWNER || membership?.role === MembershipRole.ADMIN;
+
+    if (team?.isPrivate && (!isTeamAdminOrOwner || !isOrgAdminOrOwner)) {
```

Comment:
`(team?.isPrivate && (!isTeamAdminOrOwner || !isOrgAdminOrOwner))` is true even when `isTeamAdminOrOwner` is false but `isOrgAdminOrOwner` is true


So, it should be `!isTeamAdminOrOwner && !isOrgAdminOrOwner`